// cosmwander/dist/partials/EnvList.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.envConfigs = void 0;
const jsx_runtime_1 = require("react/jsx-runtime");
const ScreenContext_1 = require("../context/ScreenContext");
exports.envConfigs = [{
  displayName: 'Osmosis Testnet',
  chainId: 'osmo-test-4',
  node: 'https://testnet-rpc.osmosis.zone:443',
  command: 'osmosisd',
  keyName: 'bob',
  feeAmount: '10',
  feeDenom: 'uosmo'
}, {
  displayName: 'Quasar Localnet',
  chainId: 'quasar',
  node: 'http://localhost:26659',
  command: 'quasarnoded',
  keyName: 'alice',
  feeAmount: '10',
  feeDenom: 'uqsr'
}];
const EnvList = ({
  height
}) => {
  const {
    env,
    setEnv
  } = (0, ScreenContext_1.useAppContext)();
  return (0, jsx_runtime_1.jsx)("box", {
    label: ' environments ',
    border: {
      type: 'line'
    },
    top: 0,
    right: 0,
    height: height,
    width: '100%',
    children: exports.envConfigs.map((conf, i) => {
      const selected = env === conf.chainId ? '**' : '';
      return (0, jsx_runtime_1.jsx)("button", {
        top: i * 3,
        height: 3,
        border: {
          type: 'line'
        },
        mouse: true,
        onPress: () => setEnv(conf.chainId),
        children: (0, jsx_runtime_1.jsx)("text", {
          children: `${selected} ${conf.displayName} ${selected}`
        })
      }, conf.chainId);
    })
  });
};
exports.default = EnvList;
