// cosmwander/dist/partials/MsgWindow.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
const jsx_runtime_1 = require("react/jsx-runtime");
const react_1 = require("react");
const ScreenContext_1 = require("../context/ScreenContext");
const fileUtils_1 = require("../utils/fileUtils");
const CodeWindow_1 = require("./MsgWindow/CodeWindow");
const ContractInstanceWindow_1 = require("./MsgWindow/ContractInstanceWindow");
const ContractWindow_1 = require("./MsgWindow/ContractWindow");
const MsgWindow = ({
  height
}) => {
  const {
    contract,
    setContract,
    codeId,
    setCodeId,
    contractInstanceAddress,
    command,
    setCommand,
    width,
    env,
    log
  } = (0, ScreenContext_1.useAppContext)();
  const [contractDetails, setContractDetails] = (0, react_1.useState)();
  (0, react_1.useEffect)(() => {
    if (contract) {
      try {
        setContractDetails((0, fileUtils_1.getContractDetails)(contract.buildName));
      } catch (e) {
        console.log(e);
      }
    }
  }, [contract, command]);
  function renderActiveWindow() {
    if (contract && !codeId) return (0, jsx_runtime_1.jsx)(ContractWindow_1.default, {
      contractDetails: contractDetails
    });else if (codeId && !contractInstanceAddress) return (0, jsx_runtime_1.jsx)(CodeWindow_1.default, {});else if (contractInstanceAddress) return (0, jsx_runtime_1.jsx)(ContractInstanceWindow_1.default, {});
  }
  return (0, jsx_runtime_1.jsx)("box", {
    label: contract?.fileName || ' msg ',
    border: {
      type: 'line'
    },
    top: 0,
    height: height,
    width: '100%',
    children: renderActiveWindow()
  });
};
exports.default = MsgWindow;
