// cosmwander/dist/partials/MsgWindow/ContractInstanceWindow.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
const jsx_runtime_1 = require("react/jsx-runtime");
const chalk = require("chalk");
const react_1 = require("react");
const ScreenContext_1 = require("../../context/ScreenContext");
const ExecuteMsgTab_1 = require("./ExecuteMsgTab");
const QueryMsgTab_1 = require("./QueryMsgTab");
const ContractInstanceWindow = () => {
  const {
    contract,
    setContract,
    codeId,
    setCodeId,
    setContractInstanceAddress,
    command,
    setCommand,
    width,
    env,
    log
  } = (0, ScreenContext_1.useAppContext)();
  const [activeTab, setActiveTab] = (0, react_1.useState)('none');
  function renderActiveTab() {
    if (activeTab === 'execute') {
      return (0, jsx_runtime_1.jsx)(ExecuteMsgTab_1.default, {});
    } else if (activeTab === 'query') {
      return (0, jsx_runtime_1.jsx)(QueryMsgTab_1.default, {});
    }
  }
  return (0, jsx_runtime_1.jsxs)("box", {
    top: 0,
    left: 0,
    children: [(0, jsx_runtime_1.jsxs)("box", {
      top: 0,
      left: 0,
      height: 3,
      children: [(0, jsx_runtime_1.jsx)("button", {
        top: 0,
        left: 0,
        width: '50%',
        height: 3,
        border: {
          type: 'line'
        },
        mouse: true,
        onPress: () => setActiveTab('execute'),
        children: activeTab === 'execute' ? chalk.inverse(`Execute`) : 'Execute'
      }), (0, jsx_runtime_1.jsx)("button", {
        top: 0,
        right: 0,
        width: '50%-1',
        height: 3,
        border: {
          type: 'line'
        },
        mouse: true,
        onPress: () => setActiveTab('query'),
        children: activeTab === 'query' ? chalk.inverse(`Query`) : 'Query'
      })]
    }), (0, jsx_runtime_1.jsx)("box", {
      top: 3,
      left: 0,
      children: renderActiveTab()
    })]
  });
};
exports.default = ContractInstanceWindow;
