// cosmwander/dist/partials/MsgWindow/QueryMsgTab.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
const jsx_runtime_1 = require("react/jsx-runtime");
const ScreenContext_1 = require("../../context/ScreenContext");
const commandUtils_1 = require("../../utils/commandUtils");
const fileUtils_1 = require("../../utils/fileUtils");
const GenericMsgTab_1 = require("./GenericMsgTab");
const QueryMsgTab = () => {
  const {
    contract,
    setContract,
    codeId,
    setCodeId,
    contractInstanceAddress,
    setContractInstanceAddress,
    command,
    setCommand,
    width,
    env,
    log
  } = (0, ScreenContext_1.useAppContext)();
  function savequeryMsg(msg, i) {
    setContract({
      ...contract,
      queryMsgs: i === null ? [...contract.queryMsgs, msg] : contract.queryMsgs.map((m, j) => {
        if (j === i) return msg;
        return m;
      })
    });
  }
  function deletequeryMsg(i) {
    setContract({
      ...contract,
      queryMsgs: contract.queryMsgs.filter((_, j) => j !== i)
    });
  }
  function sendqueryMsg(msg) {
    const envConfig = (0, commandUtils_1.getEnv)(env);
    setCommand({
      command: envConfig.command,
      args: ['query', 'wasm', 'contract-state', 'smart', contractInstanceAddress, msg.msg, '-o', 'json', '--node', envConfig.node, '--chain-id', envConfig.chainId],
      cwd: (0, fileUtils_1.getCWD)()
    });
  }
  return (0, jsx_runtime_1.jsx)(GenericMsgTab_1.default, {
    label: `Query Msg (${contractInstanceAddress.slice(0, 6)}...${contractInstanceAddress.slice(-6)})`,
    type: 'query',
    savedMsgs: contract.queryMsgs,
    saveMsg: savequeryMsg,
    deleteMsg: deletequeryMsg,
    sendMsg: sendqueryMsg
  });
};
exports.default = QueryMsgTab;
