// cosmwander/dist/MsgWindow/ExecuteMsgTab.js
"use strict";

var __importDefault = this && this.__importDefault || function (mod) {
  return mod && mod.__esModule ? mod : {
    "default": mod
  };
};
Object.defineProperty(exports, "__esModule", {
  value: true
});
const jsx_runtime_1 = require("react/jsx-runtime");
const ScreenContext_1 = require("../../context/ScreenContext");
const commandUtils_1 = require("../../utils/commandUtils");
const fileUtils_1 = require("../../utils/fileUtils");
const GenericMsgTab_1 = __importDefault(require("./GenericMsgTab"));
const ExecuteMsgTab = () => {
  const {
    contract,
    setContract,
    codeId,
    setCodeId,
    contractInstanceAddress,
    setContractInstanceAddress,
    command,
    setCommand,
    width,
    env,
    log
  } = (0, ScreenContext_1.useAppContext)();
  function saveExecuteMsg(msg, i) {
    setContract({
      ...contract,
      executeMsgs: i === null ? [...contract.executeMsgs, msg] : contract.executeMsgs.map((m, j) => {
        if (j === i) return msg;
        return m;
      })
    });
  }
  function deleteExecuteMsg(i) {
    setContract({
      ...contract,
      executeMsgs: contract.executeMsgs.filter((_, j) => j !== i)
    });
  }
  function sendExecuteMsg(msg) {
    const envConfig = (0, commandUtils_1.getEnv)(env);
    setCommand({
      command: envConfig.command,
      args: ['tx', 'wasm', 'execute', contractInstanceAddress, msg.msg, '--from', envConfig.keyName, '--gas-prices', envConfig.feeAmount + envConfig.feeDenom, '--gas', 'auto', '--gas-adjustment', '1.3', '--node', envConfig.node, '--chain-id', envConfig.chainId],
      cwd: (0, fileUtils_1.getCWD)()
    });
  }
  return (0, jsx_runtime_1.jsx)(GenericMsgTab_1.default, {
    label: `Execute Msg (${contractInstanceAddress.slice(0, 6)}...${contractInstanceAddress.slice(-6)})`,
    type: 'execute',
    savedMsgs: contract.executeMsgs,
    saveMsg: saveExecuteMsg,
    deleteMsg: deleteExecuteMsg,
    sendMsg: sendExecuteMsg
  });
};
exports.default = ExecuteMsgTab;
