// cosmwander/dist/App.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
const jsx_runtime_1 = require("react/jsx-runtime");
const ScreenContext_1 = require("./context/ScreenContext");
const Contracts_1 = require("./partials/Contracts");
const Debug_1 = require("./partials/Debug");
const EnvList_1 = require("./partials/EnvList");
const Logs_1 = require("./partials/Logs");
const MsgWindow_1 = require("./partials/MsgWindow");
const config_1 = require("./utils/config");
const windowUtils_1 = require("./utils/windowUtils");
const App = ({}) => {
  const {
    width,
    height
  } = (0, ScreenContext_1.useAppContext)();
  const firstColWidth = Math.floor(Math.min(width * 0.3, config_1.SIDEBAR_WIDTH));
  const secondColWidth = (0, windowUtils_1.getCenterColWidth)(width);
  const lastColWidth = Math.floor(Math.min(width * 0.3, config_1.SIDEBAR_WIDTH));
  const secondColLeft = firstColWidth;
  return (0, jsx_runtime_1.jsxs)("element", {
    children: [(0, jsx_runtime_1.jsx)("box", {
      width: firstColWidth,
      height: '100%',
      children: (0, jsx_runtime_1.jsx)(Contracts_1.default, {})
    }), (0, jsx_runtime_1.jsxs)("box", {
      left: secondColLeft,
      width: secondColWidth,
      height: '100%',
      children: [(0, jsx_runtime_1.jsx)(MsgWindow_1.default, {
        height: '50%'
      }), (0, jsx_runtime_1.jsx)(Logs_1.default, {
        top: '50%',
        height: '50%'
      })]
    }), (0, jsx_runtime_1.jsxs)("box", {
      right: 0,
      width: lastColWidth,
      height: '100%',
      children: [(0, jsx_runtime_1.jsx)(EnvList_1.default, {
        height: config_1.IS_DEBUG ? height - 3 : height
      }), config_1.IS_DEBUG && (0, jsx_runtime_1.jsx)(Debug_1.default, {})]
    })]
  });
};
exports.default = App;
