// cosmwander/dist/Counter.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
const jsx_runtime_1 = require("react/jsx-runtime");
const react_1 = require("react");
const Counter = props => {
  const [count, setCount] = (0, react_1.useState)(0);
  const [fontIndex, setFontIndex] = (0, react_1.useState)(0);
  const boxRef = (0, react_1.useRef)(null);
  (0, react_1.useEffect)(() => {
    const timer = setTimeout(() => {
      setFontIndex(fontIndex + 1);
      boxRef?.current.setScrollPerc(100);
    }, 1000);
    return () => clearTimeout(timer);
  }, [fontIndex]);
  function handleCountButton() {
    console.log('HERE!');
    setCount(count + 1);
  }
  const text = `You clicked ${count}\nstupid times, ${new Array(fontIndex).fill(`\nI love you ${fontIndex}`).join('')}`;
  return (0, jsx_runtime_1.jsxs)(jsx_runtime_1.Fragment, {
    children: [(0, jsx_runtime_1.jsx)("box", {
      mouse: true,
      ref: boxRef,
      scrollable: true,
      style: {
        scrollbar: {
          bg: 'blue',
          fg: 'red',
          track: {
            bg: 'cyan',
            fg: 'magenta'
          }
        }
      },
      bottom: 1,
      height: 5,
      children: text
    }), (0, jsx_runtime_1.jsx)("button", {
      top: 2,
      width: 12,
      height: 3,
      border: {
        type: 'line'
      },
      mouse: true,
      onPress: handleCountButton,
      children: "Click me"
    })]
  });
};
exports.default = Counter;
